maven-embedder/src/main/java/org/apache/maven/cli/internal/BootstrapCoreExtensionManager.java

```diff
@@ -127,30 +134,39 @@ private CoreExtensionEntry createExtension( CoreExtension extension, List<Artifa
     }
 
     private List<Artifact> resolveExtension( CoreExtension extension, RepositorySystemSession repoSession,
-                                             List<RemoteRepository> repositories, DependencyFilter dependencyFilter )
+                                             List<RemoteRepository> repositories, DependencyFilter dependencyFilter,
+                                             Interpolator interpolator )
         throws ExtensionResolutionException
     {
         try
         {
             // TODO: enhance the PluginDependenciesResolver to provide a
-            // TODO:    resolveCoreExtension method which uses a CoreExtension
-            // TODO:    object instead of a Plugin as this makes no sense
+            // TODO: resolveCoreExtension method which uses a CoreExtension
+            // TODO: object instead of a Plugin as this makes no sense
             Plugin plugin = new Plugin();
-            plugin.setGroupId( extension.getGroupId() );
-            plugin.setArtifactId( extension.getArtifactId() );
-            plugin.setVersion( extension.getVersion() );
+            plugin.setGroupId( interpolator.interpolate( extension.getGroupId() ) );
+            plugin.setArtifactId( interpolator.interpolate( extension.getArtifactId() ) );
+            plugin.setVersion( interpolator.interpolate( extension.getVersion() ) );
 
-            DependencyNode root = pluginDependenciesResolver
-                    .resolveCoreExtension( plugin, dependencyFilter, repositories, repoSession );
+            DependencyNode root =
+                pluginDependenciesResolver.resolveCoreExtension( plugin, dependencyFilter, repositories, repoSession );
             PreorderNodeListGenerator nlg = new PreorderNodeListGenerator();
             root.accept( nlg );
 
             return nlg.getArtifacts( false );
         }
-        catch ( PluginResolutionException e )
+        catch ( PluginResolutionException | InterpolationException e )
         {
             throw new ExtensionResolutionException( extension, e.getCause() );
         }
     }
 
+    private static Interpolator createInterpolatorFunction( MavenExecutionRequest request )
+    {
+        RegexBasedInterpolator interpolator = new RegexBasedInterpolator();
```

Comment:
Isn't the `StringSearchInterpolator` enough?
See `org.apache.maven.model.building.DefaultModelBuilder.interpolateModel(Model, ModelBuildingRequest, ModelProblemCollector)`

Reply:
Done.